internal/controlplane/handlers_profile.go

```diff
@@ -367,52 +369,102 @@ func (s *Server) getRuleEvaluationStatuses(
+		!dbRuleEvalStat.RemStatus.Valid ||
+		!dbRuleEvalStat.RemDetails.Valid ||
+		!dbRuleEvalStat.AlertStatus.Valid ||
+		!dbRuleEvalStat.AlertDetails.Valid {
+		return nil, fmt.Errorf("rule evaluation status not valid")
+	}
+
+	// Get the rule type info
+	ruleTypeInfo, err := store.GetRuleTypeByID(ctx, dbRuleEvalStat.RuleTypeID)
+	if err != nil {
+		l.Err(err).Msg("error getting rule type info from db")
+	} else {
+		guidance = ruleTypeInfo.Guidance
+	}
+
+	st := &minderv1.RuleEvaluationStatus{
+		ProfileId:           profileID,
+		RuleId:              dbRuleEvalStat.RuleTypeID.String(),
+		RuleName:            dbRuleEvalStat.RuleTypeName,
+		RuleTypeName:        dbRuleEvalStat.RuleTypeName,
+		RuleDescriptionName: dbRuleEvalStat.RuleName,
+		Entity:              string(dbRuleEvalStat.Entity),
+		Status:              string(dbRuleEvalStat.EvalStatus.EvalStatusTypes),
+		Details:             dbRuleEvalStat.EvalDetails.String,
+		EntityInfo:          getRuleEvalEntityInfo(ctx, store, dbEntity, selector, dbRuleEvalStat, providerName),
+		Guidance:            guidance,
+		LastUpdated:         timestamppb.New(dbRuleEvalStat.EvalLastUpdated.Time),
+		RemediationStatus:   string(dbRuleEvalStat.RemStatus.RemediationStatusTypes),
+		RemediationDetails:  dbRuleEvalStat.RemDetails.String,
+		Alert: &minderv1.EvalResultAlert{
+			Status:  string(dbRuleEvalStat.AlertStatus.AlertStatusTypes),
+			Details: dbRuleEvalStat.AlertDetails.String,
+		},
+	}
 
-		st := &minderv1.RuleEvaluationStatus{
-			ProfileId:           profileId,
-			RuleId:              dbRuleEvalStat.RuleTypeID.String(),
-			RuleName:            dbRuleEvalStat.RuleTypeName,
-			RuleTypeName:        dbRuleEvalStat.RuleTypeName,
-			RuleDescriptionName: dbRuleEvalStat.RuleName,
-			Entity:              string(dbRuleEvalStat.Entity),
-			Status:              string(dbRuleEvalStat.EvalStatus.EvalStatusTypes),
-			Details:             dbRuleEvalStat.EvalDetails.String,
-			EntityInfo:          getRuleEvalEntityInfo(ctx, s.store, dbEntity, selector, dbRuleEvalStat, providerName),
-			Guidance:            guidance,
-			LastUpdated:         timestamppb.New(dbRuleEvalStat.EvalLastUpdated.Time),
-			RemediationStatus:   string(dbRuleEvalStat.RemStatus.RemediationStatusTypes),
-			RemediationDetails:  dbRuleEvalStat.RemDetails.String,
-		}
+	if dbRuleEvalStat.RemLastUpdated.Valid {
+		st.RemediationLastUpdated = timestamppb.New(dbRuleEvalStat.RemLastUpdated.Time)
+	}
 
-		if dbRuleEvalStat.RemLastUpdated.Valid {
-			st.RemediationLastUpdated = timestamppb.New(dbRuleEvalStat.RemLastUpdated.Time)
-		}
+	if dbRuleEvalStat.AlertLastUpdated.Valid {
+		st.Alert.LastUpdated = timestamppb.New(dbRuleEvalStat.AlertLastUpdated.Time)
+	}
 
-		ruleEvaluationStatuses = append(ruleEvaluationStatuses, st)
+	// If the alert metadata is valid, parse it and set the URL
+	if dbRuleEvalStat.AlertMetadata.Valid {
+		// Define a struct to match the JSON structure
+		data := struct {
+			GhsaId string `json:"ghsa_id"`
```

Comment:
can we move this to a function so we don't hardcode github-specific pieces into this logic?

Reply:
good point 👍 move it out to a separate function.